extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/Util.java

```diff
@@ -0,0 +1,105 @@
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.types.State;
+
+/**
+ * The {@link Util} is a set of helper functions
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class Util {
+
+    /**
+     * convert a number to a list of booleans
+     *
+     * @param decimal
+     * @return
+     */
+    public static List<Boolean> decimalTypeToBooleanList(DecimalType decimal) {
+        List<Boolean> returnValues = new ArrayList<Boolean>();
+        int value = decimal.intValue();
+        for (int i = 0; i < 8; i++) {
+            returnValues.add((value & (1 << i)) > 0);
+        }
+        return returnValues;
+    }
+
+    /**
+     * convert measured value to temperature with offset
+     *
+     * @param temperature measured value
+     * @param offset offset to apply
+     * @return originalState if not DecimalType, temperature otherwise
+     * @throws OwException in case of conversion error
+     */
+    public static QuantityType<Temperature> convertTemperature(QuantityType<Temperature> temperature,
+            QuantityType<Temperature> offset) throws OwException {
+        QuantityType<Temperature> returnTemperature = temperature.toUnit(offset.getUnit());
+        if (returnTemperature == null) {
+            throw new OwException("error while converting measured value");
+        }
+        returnTemperature = returnTemperature.add(offset).toUnit(SIUnits.CELSIUS);
+        if (returnTemperature == null) {
+            throw new OwException("error while adding offset");
+        }
+        return returnTemperature;
+    }
+
+    /**
+     * calculate absolute humidity in g/m³ from measured values
+     *
+     * @param temperature the measured temperature
+     * @param relativeHumidity the measured relative humidity
+     * @return the corresponding absolute humidity
+     */
+    public static State calculateAbsoluteHumidity(QuantityType<Temperature> temperature,
+            QuantityType<Dimensionless> relativeHumidity) {
+        Double theta = temperature.toUnit(SIUnits.CELSIUS).doubleValue();
+        // saturation vapor pressure in kg/(m s^2)
+        Double saturationVaporPressure = 611.2 * Math.exp(17.62 * theta / (243.12 + theta));
+        // absolute humidity in kg/m^3
+        Double aH = relativeHumidity.doubleValue() / 100 * saturationVaporPressure / (461.52 * (273.15 + theta));
+        State absoluteHumidity = new QuantityType<Density>(aH, SmartHomeUnits.KILOGRAM_PER_CUBICMETRE).toUnit("g/m³");
+        if (absoluteHumidity != null) {
+            return absoluteHumidity;
+        } else {
+            throw new IllegalArgumentException("could not change unit");
+        }
+    }
+
+    public static State calculateDewpoint(QuantityType<Temperature> temperature,
```

Comment:
Also nice! See my last comment about derived values. No need to change anything right now :-)